crates/pow/src/challenge_cache.rs

```diff
@@ -25,6 +25,8 @@ pub(crate) struct ChallengeCache {
     challenges_per_key: HashMap<ApiKey, usize>,
     /// Maps account id to the number of submitted challenges.
     account_ids: BTreeMap<AccountId, usize>,
+    /// Maps account id and api key to the timestamp of the last submitted challenge.
+    challenges_timestamps: BTreeMap<(AccountId, ApiKey), u64>,
```

Comment:
This is not wrong per se, but the usual default is to use a HashMap. Are we using a BTreeMap for some reason?